src/QsCompiler/SyntaxProcessor/CapabilityInference/FeatureAnalyzer.fs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+module Microsoft.Quantum.QsCompiler.SyntaxProcessing.CapabilityInference.FeatureAnalyzer
+
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.Diagnostics
+open Microsoft.Quantum.QsCompiler.SyntaxProcessing.CapabilityInference
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Transformations.Core
+
+type Feature =
+    | Fail
+    | IndeterminateLoop
+    | MultipleReturns
+    | NontrivialCallee
+    | DefaultArray
+
+let createPattern range feature =
+    let capability = TargetCapability.withClassical ClassicalCapability.full TargetCapability.bottom
+
+    let diagnose (target: Target) =
+        let range = QsNullable.defaultValue Range.Zero range
+
+        if TargetCapability.subsumes target.Capability capability then
+            None
+        else
+            let description =
```

Comment:
Same comment regarding the split of the diagnostic text here as on the other file.

Reply:
Resolving as per the discussion above.